### DISCUSSION TITLE
Is tere option to make image tile / seamless?

### DISCUSSION THREAD

**Discussion by JoseConseco** [Acly/krita-ai-diffusion#938] [Q&A]
I know automatic 1111 had this option to make image seamless at the border, it was great for creating repeating textures. Is there  option     for tiling image gen in krita-ai?  I'm not sure how comfy - to krita works, but I googled some comfy setups for making tiled images: https://github.com/spinagon/ComfyUI-seamless-tiling  - maybe this can be integreated somehow in comfy krita ui?

**Comment by D-side** [comment]
For whatever it's worth, here's the file responsible for the same feature in InvokeAI: https://github.com/invoke-ai/InvokeAI/blob/main/invokeai/backend/stable_diffusion/seamless.py

Notably, it includes two separate switches for horizontal and vertical tiling, which would be nice to have here as well.

**Comment by Drakosha405** [comment]
> I know automatic 1111 had this option to make image seamless at the border, it was great for creating repeating textures. Is there option for tiling image gen in krita-ai? I'm not sure how comfy - to krita works, but I googled some comfy setups for making tiled images: https://github.com/spinagon/ComfyUI-seamless-tiling - maybe this can be integreated somehow in comfy krita ui?

why do you have to do this in krita? it's easier to create it in the comfy ui and then just send it to krita.
or do it manually. create a texture -- shift along the axes -- highlight seams -- create patches.

**Reply by D-side** [reply]
This should probably be considered a feature request rather than a discussion question.
It's only easier to do through ComfyUI precisely because this feature apparently doesn't exist. And convincing an artist to grasp ComfyUI is generally a losing proposition.

Unfortunately, no, it isn't as simple as generating an image and filling the seams, as proper seamless generation affects the general composition of the image such as avoiding obvious use of perspective (at least that's been my experience with InvokeAI).

**Reply by Drakosha405** [reply]
I considered it from this point of view. if you need seamless at once. you probably won't change anything manually. So you can make hundreds of samples at once by saving them and it's easier to transfer this texture from the Comfy ui to the blender for an example.if there is a need to change something manually, you will have a choice from a large number of images.

**Comment by JoseConseco** [comment]
Right now I use the solution from my first post, I it would be just cool to have this directly in Krita (eg. after enabling canvas tiling in krita,  krita-ai  would enable these  tiling nodes automatically):
![image](https://github.com/user-attachments/assets/9a86c4b3-38aa-4a36-b7ee-816e878d2376)

I will make issue with feature request.  DONE: https://github.com/Acly/krita-ai-diffusion/issues/956
